WalletWasabi.Documentation/Guides/Wasabi_Privacy_Considerations.md

```diff
@@ -0,0 +1,47 @@
+# Wasabi Wallet Privacy Considerations
+
+> _Known limitations and how to prevent de-anonymizing yourself while using Wasabi. Note that this list is not comprehensive and other limitations or weaknesses may be found in the future._
+
+### 1. Neutrino is not as private as using one's own full node
+
+The Neutrino protocol used by Wasabi for UTXO confirmation is better for privacy than SPV but still not as private as using one's own full node as the UTXO provider.
```

Comment:
I think we should not make references to Neutrino in this way. We should be more precise and say BIP158 or Client-side filters. This is because: 1) Neutrino is just the name of another wallet, it used BIP158 and BIP157 (we do not use BIP157), We use compact filter with different parameters and only for native segwit transactions, etc.

Reply:
Re: Neutrino vs BIP158, good point, I will change that if we decide to keep this section.

Re: Overall privacy claims, I stand by the claim that using one's own full node is more private than BIP158. I tend to agree with Peter Wuille's contribution to the linked reddit thread:

> See it this way: inherently, any method that only downloads a selection of transactions of blocks, where that selection can be partially observed, will teach the observer something. This is even the case without address reuse - an attacker could see see a few blocks being fetched of transactions that are related through the spending graph, and at least conclude there is now an elevated probability they're related to the node fetching them. To increase privacy you can increase the blocks you fetch randomly, or spread things out over more peers. But these approaches come at a cost, and still only weaken the signal... So no, there absolutely still is an information leak. It is just dramatically weaker, and we should be happy about that. But if you actually want no leak at all, you'll either need to download every block, or use cryptographic techniques to make queries to a server without it learning what it was being asked about (private information retrieval, PIR).

https://www.reddit.com/r/Bitcoin/comments/9agftj/merged_pr_from_jimpo_bip_158_compact_block/e4wldur/

Can you provide documentation/ evidence to back up the claim that "there is no difference" between your BIP158 implementation and using one's own full node? I'll need something concrete to point to to feel comfortable removing this section.

Reply:
It's like pure math vs crypto. From a mathematical point of view we know that `2+2=4`, but we also know that `pubkey -> privkey` is possible, it's just extremely unlikely.  

The above conversation was a back and forth: 

- me: `pubkey -> privkey` is extremely unlikely.
- other party: sure, but `pubkey -> privkey` is theoretically possible.
- me: but it's unlikely.
- other party: but it's possible.
- ...

Sometimes I think about that we might have fearmongered too much even over bloom filtering SPV wallets and @jonasnick's issues could've been reasonably addressed in more straightforward ways, like @NicolasDorier's https://github.com/NicolasDorier/NBitcoin.SPVSample. Anyhow, fearmongering over BIP158 without real basis is not only unreasonable, it's highly counterproductive.

Reply:
@nopara73 thank you for the explanation. Using your metaphor, I can prove that a user's own full node will leave no data about their wallet addresses in aggregate to third parties (2+2=4). However, I cannot prove the same to the user if they use BIP-158. Rather, I can tell them the probability is very low that it could happen (pubkey -> privkey). So it would be accurate to say that the former is more private than the latter. 

I don't think this is fearmongering, it is just stating the facts. The question might be however whether it is worth pointing out this fact or whether it is TMI for end users to worry about (e.g. I don't remember reading about quantum crypto attacks in GPG documentation). If it is your opinion that we are dealing with the latter case then I will remove this section of the documentation and we will have this conversation to point to if there is ever a question about it later on.